src/service/user-service.js

```diff
@@ -0,0 +1,161 @@
+const mongoSvc = require('./mongo-service');
+const User = require('../model/user-model');
+
+const QueryBuilder = function() {
+  const query = {}
+  return {
+    id: function(id) {
+      if (id) {
+        query['id'] = id
+      }
+      return this
+    },
+    name:function(name) {
+      if (name) {
+        query['name'] = name
+      }
+      return this
+    },
+    build: function() {
+      return query
+    }
+  }
+};
+
+const svc = {
+
+  /**
+   * retrieve users from db
+   * @param query {{ id: String, name: String }}
+   * @param option {{ page: number, limit: number }}
+   * @param logger { Pino }
+   * @return {{ docs: Array, total: number }}
+   */
+  getUser: async function(query, option, logger = log) {
+    const _log = logger.child({ module: 'user-service', method: 'getUser' });
+    _log.info('retrieve user, query: %j, option: %j', query, option);
+
+    const dbConnection = mongoSvc.getMongoConnection(_log);
+    if (dbConnection == null) {
+      _log.error('Cannot get mongodb connection.');
+      throw new errors.InternalServerError('Cannot get mongodb connection.');
```

Comment:
The mongodb connection error handling seems to appear in many places in this user-service,
an improvement is expected.

Reply:
Sure. I will improve it in next commit. Thanks  for your suggestion.